ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -13,6 +13,151 @@ namespace ext
 namespace MitsubaLoader
 {
 
+static uint64_t rgb32f_to_rgb19e7(const uint32_t _rgb[3])
+{
+	constexpr uint32_t mantissa_bitlen = 19u;
+	constexpr uint32_t exp_bitlen = 7u;
+	constexpr uint32_t exp_bias = 63u;
+
+	constexpr uint32_t mantissa_bitlen_f32 = 23u;
+	constexpr uint32_t exp_bitlen_f32 = 8u;
+	constexpr uint32_t exp_bias_f32 = 127u;
+
+	constexpr uint32_t mantissa_len_diff = mantissa_bitlen_f32 - mantissa_bitlen;
+	constexpr uint32_t exp_bias_diff = exp_bias_f32 - exp_bias;
+
+	uint64_t rgb19e7 = 0ull;
+	for (uint32_t i = 0u; i < 3u; ++i)
+	{
+		uint64_t mantissa = _rgb[i] & 0x7fffffu;
+		mantissa >>= mantissa_len_diff;
+		mantissa <<= (mantissa_bitlen * i);
+		rgb19e7 |= mantissa;
+	}
+	uint64_t exp = (_rgb[0] >> mantissa_bitlen_f32) & 0xffu;
+	exp -= exp_bias_diff;
+	rgb19e7 |= (exp << (3u * mantissa_bitlen));
+
+	return rgb19e7;
+}
+
+static bsdf::SBSDFUnion bsdfNode2bsdfStruct(const CElementBSDF* _node)
+{
+	bsdf::SBSDFUnion retval;
+	uint32_t bitfields = _node->type;
+	switch (_node->type)
+	{
+	case CElementBSDF::Type::DIFFUSE:
+		_IRR_FALLTHROUGH;
+	case CElementBSDF::Type::ROUGHDIFFUSE:
+		bitfields |= static_cast<uint32_t>(_node->diffuse.reflectance.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_SHIFT_REFL_TEX;
+		bitfields |= static_cast<uint32_t>(_node->diffuse.alpha.value.type==SPropertyElementData::INVALID)<<bsdf::BITFIELDS_MASK_ALPHA_U_TEX;
+		if (_node->diffuse.reflectance.value.type==SPropertyElementData::SPECTRUM)
+			retval.diffuse.reflectance = rgb32f_to_rgb19e7(reinterpret_cast<const uint32_t*>(_node->diffuse.reflectance.value.vvalue.pointer));
+		if (_node->diffuse.alpha.value.type==SPropertyElementData::FLOAT)
+			reinterpret_cast<float&>(retval.diffuse.alpha[0]) = _node->diffuse.alpha.value.fvalue;
```

Comment:
add a `intBitsToFloat`,`uintBitsToFloat`, `floatBitsToInt` and `floatBitsToUint` to the `irr::core::` namespace as template functions in `glslFunctions.h` and `.tcc`

Reply:
still an issue

Reply:
done